uint32 handle
---
int32 result

